ql/src/java/org/apache/hadoop/hive/ql/parse/CalcitePlanner.java

```diff
@@ -1286,6 +1289,181 @@ private void fixUpASTAggregateIncrementalRebuild(ASTNode newAST) throws Semantic
     ctx.addDestNamePrefix(2, Context.DestClausePrefix.INSERT);
   }
 
+  private void fixUpASTJoinIncrementalRebuild(ASTNode newAST) throws SemanticException {
```

Comment:
Maybe these related `fixUp*` methods should be moved outside of this class (`HiveMaterializedViewUtils`?). Feel free to do it in a follow-up JIRA though.

Reply:
I'll do it in a follow-up Jira with other refactoring. I was thinking about moving these to `AlterMaterializedViewRebuildAnalyzer` somehow.